packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -116,7 +117,9 @@ export class Snapshot {
           ResourceSnapshotsReportStatus.InProgress,
         ].includes(this.latestReport.status);
 
-        if (isNotDone) {
+        const isRightOperationType = this.latestReport.type === type;
+
+        if (isNotDone || !isRightOperationType) {
```

Comment:
I would check for `isRightOperationType` before trying to initialize `isNotDone`, and throw if it ain't the rightOperationType (although I think tbh that having a variable here is really too much 😅 )

So something akin to
```ts
if (this.latestReport.type !== type) {
  throw new Error('Snapshot is still being processed'); // or a more appropriate message maybe?
}

const isNotDone = [
  ResourceSnapshotsReportStatus.Pending,
  ResourceSnapshotsReportStatus.InProgress,
].includes(this.latestReport.status);

if (isNotDone) {
  throw new Error('Snapshot is still being processed');
}
```

The boon here is that you save a declaration and an array manipulation.

If you want to really polish off this, you could even:
 - Declare a private static member containing the status.
 
 You would save the declaration and also, you would clean up the code:
 
```ts
private static ongoingReportStatuses = [
  ResourceSnapshotsReportStatus.Pending,
  ResourceSnapshotsReportStatus.InProgress,
];
/* .... */
/* .... */
/* .... */
public async waitUntilDone(type: ResourceSnapshotsReportType) {
  const waitPromise = backOff(
    async () => {
      await this.refreshSnapshotData();
      
      if (this.latestReport.type !== type) {
        throw new Error('Snapshot is still being processed'); // or a more appropriate message maybe?
      }

      if (Snapshot.ongoingReportStatutes.includes(this.latestReport.status)) {
        throw new Error('Snapshot is still being processed');
      }
    }
  )
}
```

Reply:
👌